ledger-api/grpc-definitions/com/digitalasset/admin/api/v1/package_management_service.proto

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.digitalasset.admin.api.v1;
+
+option java_outer_classname = "PackageManagementServiceOuterClass";
+option java_package = "com.digitalasset.amin.api.v1";
+
+// Allows clients to query the DAML-LF packages that are supported by the server.
+service PackageManagementService {
+
+  // Returns the details of all supported packages.
+  rpc ListPackageDetails (ListPackageDetailsRequest) returns (ListPackageDetailsResponse);
+
+  // Uploads a DAR archive.
+  rpc UploadArchive (UploadArchiveRequest) returns (UploadArchiveResponse);
+
+}
+
+message ListPackageDetailsRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification Service.
+  // Required
+  string ledger_id = 1;
+}
+
+message ListPackageDetailsResponse {
+
+  // The IDs of all DAML-LF packages supported by the server.
+  // Required
+  repeated string PackageDetails = 1;
+}
+
+message PackageDetails {
+ 
+  // The IDs of all DAML-LF packages supported by the server.
+  // Required
+  string package_id = 1;
+
+  // Human readable name of the package.
+  // Optional
+  string package_name = 2;
```

Comment:
Where is this supposed to come from? The `filename` of the DAR that contained this DALF?

Reply:
What is this set to for e.g. the daml-prim.dalf? Does the first uploader get to decide how a package is named? Does not seem right to me.
If we need human readable names, then that should really be part of the .dalf format and part of the hashing scheme.

Reply:
I have put this in here as an afterthought, after having seen it in canton version of the interface. I put it here hear the opinions about ideas floated in this context elsewhere.